ddtrace/tracer/option.go

```diff
@@ -289,7 +284,20 @@ func newConfig(opts ...StartOption) *config {
 	if c.debug {
 		log.SetLevel(log.LevelDebug)
 	}
-	c.loadAgentFeatures()
+	if info, err := agent.LoadFeatures(c.agentAddr, c.httpClient); err == nil {
```

Comment:
❓ Did you consider a more object-oriented approach for encapsulating the agent? It seems like "an agent" is essentially the combination of an agent addr and http client, i.e.:

```
type Agent struct {
  Addr string
  Client *http.Client
}
```

`LoadFeatures` could then become a method on this struct. I realize that there are cases where accessing the client or addr directly might still exist, so initially those fields could be public like shown above.